### PR TITLE
Compensation(Dummy for test) and Withdraw Compensation

### PR DESCRIPTION
**Pull request by Takunari**
I pushed the "Withdraw Compensation contract". PLS take a look at the sources. Thanks

### CODE REVIEW

File: src/contract-v2/xWinRobustCompensation.sol [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+
+    struct UserInfo {
+        uint compensateBalance;
+    }
+    
+    string public name;
+
+    /* Commented out for Test Net
+    // Token Addresses in Real Net
+    address public targetToken = address(0xd88ca08d8eec1E9E09562213Ae83A7853ebB5d28);
```
Comment (Jolly-Walker) [inline src/contract-v2/xWinRobustCompensation.sol:701]: uncomment this line, use mainnet token

File: src/contract-v2/xWinRobustCompensation.sol [hunk 2]
```diff
@@ -0,0 +1,758 @@
...
+
+        require(compensateBalance > 0, "no compensate balance");
+
+        require(_xWinCompensation.releaseBlock() < block.number, "not yet release");
+        uint256 xwinTokenBal = IBEP20(targetToken).balanceOf(address(this));
+
+        require(xwinTokenBal >= compensateBalance, "not enough to transfer");
+        TransferHelper.safeTransfer(targetToken, msg.sender, compensateBalance);
+
+        hasWithdrawed[msg.sender] = true;
```
Comment (Jolly-Walker) [inline src/contract-v2/xWinRobustCompensation.sol:745]: move this line of code 754 to before the transfer call at line 743

**Review by Jolly-Walker — CHANGES_REQUESTED** [review]

File: src/contract-v2/xWinRobustCompensation.sol (hunk 1 above)
Comment (Josezawa) [reply src/contract-v2/xWinRobustCompensation.sol:701]: Ok,
should I change the address of the xWinCompensation?
`   xWinCompensation _xWinCompensation = xWinCompensation(address(0xB722B10c219442501f60d014FCD41A9C7fe65B75));`
currently I'm using address of the xWinCompensation on the test net.

File: src/contract-v2/xWinRobustCompensation.sol (hunk 2 above)
Comment (Josezawa) [reply src/contract-v2/xWinRobustCompensation.sol:745]: OK,
I wondered whether the user could withdraw or could not withdraw anymore if the status changed before the transfer and the transfer failed.

**Comment by Josezawa** [comment]
[Jolly-Walker](https://github.com/Jolly-Walker),  PLS  review the following changes. Thanks
[1a23eb7](https://github.com/xwinfinance/xWinBSC/pull/1/commits/1a23eb7061099728a3624d2b4224ec6a86a81fd0)

**Review by xwinfinance — APPROVED** [review]

**Pull request merged**